logstash_async/transport.py

```diff
@@ -3,15 +3,63 @@
 # This software may be modified and distributed under the terms
 # of the MIT license.  See the LICENSE file for details.
 
+from abc import ABC, abstractmethod
+from time import sleep
+import json
+import random
 import socket
 import ssl
 import sys
 
+from requests.auth import HTTPBasicAuth
 import pylogbeat
+import requests
 
 from logstash_async.utils import ichunked
 
 
+class Transport(ABC):
+    """The :class:`Transport <Transport>` is the abstract base class of
+    all transport protocols.
+
+    :param host: The name of the host
+    :type host: str
+    :param port: The port number of the service
+    :type port: int
+    :param timeout: The timeout for the connection
+    :type timeout: float
+    :param ssl_enable: Use TLS for the transport (Default: True)
+    :type ssl_enable: bool
+    :param ssl_verify: If True the class tries to verify the TLS certificate
+    with certifi. If you pass a string with a file location to CA certificate
+    the class tries to validate it against it. (Default: True)
+    :type ssl_verify: bool or str
+    """
+
+    def __init__(
+            self,
+            host,
+            port,
+            timeout,
+            ssl_enable=True,
+            ssl_verify=True
+    ):
+        self.host = host
```

Comment:
Why not making the instance attributes private or in this maybe better semiprivate?
Depending on your decision, also for `HttpTransport.username` and `HttpTransport.password`.

Reply:
Of course the attributes should be private but then I would need public functions to access the variables. Without properties or getters/setters I would prefer protected.

Reply:
When I say "semiprivate" I mean what you mean by "protected", just to be sure we are talking about the same.

Do you need to access the instance attributes from outside? If I didn't miss anything, they are accessed only in the base and sub class? So semiprivate/protected should be fine.